File: python-flask-api/app/models/students.py
from app import db
from datetime import datetime
from wtforms import Form, StringField
from wtforms.validators import *
from .basemodel import BaseModel
from ..helpers.utils import OptionalButNotEmpty


class Students(db.Model):
    __tablename__ = 'students'
    id = db.Column("id", db.Integer, primary_key=True, autoincrement=True)
    fullname = db.Column("fullname", db.String)
    dob = db.Column("dob", db.DateTime)
    gender = db.Column("gender", db.String)
    residence = db.Column("residence", db.String)
    phone = db.Column("phone", db.String)
    email = db.Column("email", db.String)
    
    def _asdict(self):
       return {c.name: getattr(self, c.name) for c in self.__table__.columns}


class StudentsAddForm(Form):
    fullname = StringField('Fullname', [InputRequired()])
    dob = StringField('Dob', [InputRequired()])
    gender = StringField('Gender', [InputRequired()])
    residence = StringField('Residence', [InputRequired()])
    phone = StringField('Phone', [InputRequired()])
    email = StringField('Email', [InputRequired(),Email()])


class StudentsEditForm(Form):
    fullname = StringField('Fullname', [OptionalButNotEmpty()])
    dob = StringField('Dob', [OptionalButNotEmpty()])
    gender = StringField('Gender', [OptionalButNotEmpty()])
    residence = StringField('Residence', [OptionalButNotEmpty()])
    phone = StringField('Phone', [OptionalButNotEmpty()])
    email = StringField('Email', [OptionalButNotEmpty(),Email()])

File: nodejs-express-api/models/students.js

import { BaseModel, sequelize, Sequelize } from "./basemodel.js";

class Students extends BaseModel {
	static init() {
		return super.init(
			{
				
				id: { type: Sequelize.INTEGER, primaryKey: true, autoIncrement: true },
				fullname: { type:Sequelize.STRING , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				dob: { type:Sequelize.DATEONLY , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				gender: { type:Sequelize.STRING , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				residence: { type:Sequelize.STRING , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				phone: { type:Sequelize.STRING , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				email: { type:Sequelize.STRING , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				class: { type:Sequelize.STRING   }
			}, 
			{ 
				sequelize,
				
				tableName: "students",
				modelName: "students",
			}
		);
	}
	
	static listFields() {
		return [
			'fullname', 
			'gender', 
			'class', 
			'id'
		];
	}

	static viewFields() {
		return [
			'fullname', 
			'dob', 
			'gender', 
			'residence', 
			'phone', 
			'email', 
			'class', 
			'id'
		];
	}

	static editFields() {
		return [
			'fullname', 
			'dob', 
			'gender', 
			'residence', 
			'phone', 
			'email', 
			'class', 
			'id'
		];
	}

	
	static searchFields(){
		return [
			Sequelize.literal("fullname LIKE :search"), 
			Sequelize.literal("gender LIKE :search"), 
			Sequelize.literal("class LIKE :search"), 
			Sequelize.literal("residence LIKE :search"), 
			Sequelize.literal("phone LIKE :search"), 
			Sequelize.literal("email LIKE :search"), 
			Sequelize.literal("id LIKE :search"),
		];
	}

	
	
}
Students.init();
export default Students;
